lib/segment/src/common/operation_time_statistics.rs

```diff
@@ -82,6 +91,7 @@ impl std::ops::Add for OperationDurationStatistics {
                 other.max_duration_micros,
                 |a, b| a > b,
             ),
+            last_responded: std::cmp::max(self.last_responded, other.last_responded),
```

Comment:
You can use `self.last_responded.max(other.last_responded)`, could be a bit nicer.